### PR TITLE
Fix handling of branch names containing forward slashes

### LINKED ISSUE #3882: GitHub rev resolution fails if branch name contains forward slashes
1. Visit https://github.com/sourcegraph/sourcegraph/blob/bl/wip/dev/CLA.txt
2. Witness "Configure Sourcegraph" instead of "View Repository on Sourcegraph" button
3. Witness "View file on Sourcegraph" links to a broken location (https://sourcegraph.com/github.com/sourcegraph/sourcegraph@bl/-/blob/wip/dev/CLA.txt?diff=87a76b38af1dcb657013387504240cc850a8487a), which 404s on sourcegraph.com

This is because our GitHub `parseURL()` function relies on DOM inspection (!) through [`getBranchName()`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/browser/src/libs/github/util.tsx#L250-268) when the branch name contains forward slashes, and the selectors in `getBranchName()` are currently broken. Additionally, I'm pretty sure that even with working selectors there's other bugs lurking there...

### PR DESCRIPTION
**Pull request by lguychard**
Fixes #3882

Refactors GitHub `parseURL()` by simplifying the returned `GitHubURL` type. We now no longer attempt to parse the rev and the filePath from the URL, they must be scraped from the DOM.

##### Code Hosts

- [x] GitHub
- [x] GitHub Enterprise
- [ ] Refined GitHub
- [ ] Phabricator
- [ ] Phabricator integration
- [ ] Bitbucket
- [ ] Gitlab

##### Browsers

- [x] Chrome
- [ ] Firefox

### CODE REVIEW

File: browser/src/libs/github/util.tsx [hunk 1]
```diff
@@ -180,93 +180,70 @@ function getDiffResolvedRevFromPageSource(pageSource: string, isPullRequest: boo
...
-        rev,
-        filePath,
-        ghRepoName,
-        isDelta,
-        isPullRequest,
-        position,
-        isCommit,
-        isCodePage,
-        isCompare,
+export function isDeltaPageType(pageType: GitHubURL['pageType']): boolean {
```
Comment (felixfbecker) [inline browser/src/libs/github/util.tsx]: I find the term "delta" really weird cause it's not used anywhere else. How about "diff" instead?

Comment (lguychard) [reply browser/src/libs/github/util.tsx]: Yup, that would work. I think I let myself be influenced by the old variable name (`isDelta`)

**Comment by felixfbecker** [comment]
What's blocking this?

**Comment by lguychard** [comment]
It was based on #3920, I'll rebase and merge now.

**Pull request merged**